vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -305,7 +305,7 @@ void calculate_and_insert_interactions(
           !found_namespaces[action_index - INTERACTIONS::printable_start])
       {
         found_namespaces[action_index - INTERACTIONS::printable_start] = true;
-        generated_interactions.push_back({(char)action_index, (char)ccb_id_namespace});
+        generated_interactions.push_back({static_cast<namespace_index>(action_index), static_cast<namespace_index>(ccb_id_namespace)});
```

Comment:
Instead of static_casting the `action_index`, we should just make it the correct type

https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/example_predict.h#L6
We have at least 2 different definitions for what a `namespace_index` is. We need to track any additional definitions down and unify everything

Reply:
We don't need a static cast here. I'll remove that.

Also, `example_predict.h` defines `namespace_index` as an `unsigned char`